src/cli/subcommands/snapshot_cmd.rs

```diff
@@ -196,11 +206,52 @@ impl SnapshotCommands {
 
             Self::Compress {
                 source,
-                destination,
+                output,
                 compression_level,
                 frame_size,
+                force,
             } => {
-                let file = tokio::io::BufReader::new(File::open(&source).await?);
+                // If input is 'snapshot.car.zst' and output is '.', set the
+                // destination to './snapshot.forest.car.zst'.
+                let destination = match output.is_dir() {
+                    true => {
+                        let mut destination = output;
+                        destination.push(source.clone());
+                        while let Some(ext) = destination.extension() {
+                            if !(ext == "zst" || ext == "car" || ext == "forest") {
+                                break;
+                            }
+                            destination.set_extension("");
+                        }
+                        destination.with_extension("forest.car.zst")
+                    }
+                    false => output.clone(),
+                };
+
+                if destination.exists() && !force {
+                    let have_permission = Confirm::with_theme(&ColorfulTheme::default())
+                        .with_prompt(format!(
+                            "{} will be overwritten. Continue?",
+                            destination.to_string_lossy()
+                        ))
+                        .default(false)
+                        .interact()
+                        // e.g not a tty (or some other error), so haven't got permission.
+                        .unwrap_or(false);
+                    if !have_permission {
+                        return Ok(());
+                    }
+                }
+
+                println!("Generating ForestCAR.zst file: {:?}", &destination);
+
+                let file = File::open(&source).await?;
+                let pb = ProgressBar::new(file.metadata().await?.len()).with_style(
```

Comment:
Just to confirm, should we only use log based progress bar in daemon, and `indicatif` in cli and tools?

Reply:
Yes.